axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/axon/PersistentStreamConnection.java

```diff
@@ -128,12 +130,18 @@ public PersistentStreamConnection(String streamId,
 
 
     /**
-     * Initiates the connection to Axon Server to read events from the persistent stream.
+     * Initiates the connection to Axon Server to read events from the persistent stream. The stream can have only one
+     * consumer. If you invoke the method multiple times, the previous consumer is replaced with the new one.
      *
      * @param consumer The consumer of batches of event messages.
      */
     public void open(Consumer<List<? extends EventMessage<?>>> consumer) {
-        this.consumer.set(consumer);
+        Consumer<List<? extends EventMessage<?>>> previousConsumer = this.consumer.getAndSet(consumer);
+        if (!previousConsumer.equals(NO_OP_CONSUMER)) {
```

Comment:
IMO we should prevent that (not just warn), because in `start()` we open a new persistent stream and after the replacement we won't be able to close it.

Reply:
I've given it some thought, and I think it would make sense indeed. 

The persistent streams are intended to go to a specific event processor. 
If another event processor accidentally takes over, that means the trace within the stream switches too. It would thus cause partial processing of events on different event processors, making their query models or processes act on partial events.

Although we should block it in the connection, I think we should also block it in the `PersistentStreamMessageSource`.

Reply:
I would vote to throw an exception when the `open(...)` method is called a second time. Once a consumer has been registered, the connection should be exclusive to that consumer.
We can do this atomically with a `this.consumer.compareAndSet()`.

Reply:
@smcvb 
The `SubscribableMessageSource` docs for subscribe says `If the given {@code messageProcessor} is already subscribed, nothing happens.` What should we do with that? 
Check if the consumer already subscribed is the same? If not - throw exception, if yes - just do nothing (return existing Registration)>